### ISSUE TITLE
SQLCipher .NET 8 / MAUI Issues on iOS Release

### ISSUE THREAD

**Issue by phillippschmedt** [praeclarum/sqlite-net#1207]
Hi there,

we run into issues that reads and writes to the database are not working when running our iOS MAUI App in release mode. In Debug Mode everything is fine. We do not get any exceptions!

Packages:
        <PackageReference Include="sqlite-net-sqlcipher" Version="1.8.116" />
        <PackageReference Include="SQLitePCLRaw.core" Version="2.1.5" />
        <PackageReference Include="SQLitePCLRaw.provider.sqlite3" Version="2.1.5" />
        <PackageReference Include="SQLitePCLRaw.bundle_green" Version="2.1.5" />
        <PackageReference Include="SQLitePCLRaw.provider.dynamic_cdecl" Version="2.1.5" />
        <PackageReference Include="SQLitePCLRaw.bundle_e_sqlcipher" Version="2.1.5" />

Is anyone experiencing the same issues or has any idea how we can fix this?

Thanks a lot

**Comment by jurganson** [comment]
Bump!

**Comment by jurganson** [comment]
Yeah seems like 1.7.335 is the way to go

**Comment by pme442** [comment]
bump

**Comment by rdavisau** [comment]
Use these settings in Release:

```
        <UseInterpreter>true</UseInterpreter>
        <MtouchInterpreter>-all</MtouchInterpreter>
```

This means, "enable the interpreter, but don't use it on any project assemblies". 
Your app's assemblies will be AOT compiled as usual, but the interpreter will be available at runtime to handle any dynamically generated code.

**Comment by mjo151** [comment]
The problem with running on iOS was caused by the addition of fast column setters #902. You need to remove or #ifdef the code that was added for fast column setters and it should work fine on iOS without the interpreter.

**Comment by BlueRaja** [comment]
Xamarin End-of-life is in 3 months.  After that, it won't be long before Xamarin apps no longer even build.  This needs to be addressed soon or this library is effectively dead.

**Comment by Zhangos** [comment]
Is there another way?

**Comment by nbevans** [comment]
> The problem with running on iOS was caused by the addition of fast column setters #902. You need to remove or #ifdef the code that was added for fast column setters and it should work fine on iOS without the interpreter.

This is the way. Isn't there new AOT-compatible techniques for doing this type of thing as well? What does Dapper do?

**Comment by phillippschmedt** [comment]
I would like to bum this issue. It's still relevant. As of now I copied over the source code of this project and removed the FastColumnSetter.

**Comment by IVSoftware** [comment]
As a "MAUI developer who develops for iOS" having this exact issue, I had tried various things over several days (because release mode takes 10-15 minutes to build on my setup) and, fortunately, took a tack today that led me to this thread and the viable workaround. The comment @ckrutsinger made me consider whether or not there was an exception to be caught, and that I had laid some groundwork and might be in a good position to capture it if there was. I should also mention that while this thread is tagged as .NET 8, my project is currently building on .NET 7 (a distinction that doesn't appear to have any bearing).

What I can offer as a data point is this:

```csharp
private async void onStartupTaskG(object sender, CheckedChangedEventArgs e)
{
    using (var cnx = new SQLiteConnection(SpecialPath.Sqm.GetRelEnvPath()))
    {
        Profile.Log($"YVKHU");
        var acnxSqm = ACnx[SpecialPath.Sqm];
        try
        {
            List<DBPI> sqmDbpisA = await
                acnxSqm
                .Table<DBPI>()
                .ToListAsync();
            Profile.Log($"{NewLine}{string.Join(NewLine, sqmDbpisA.Select(_=>_.description))}");
            Profile.Log($"SNRZ3");
        }
        catch (Exception ex)
        {
            Profile.Log($"NPGBK: {ex.GetType().Name}{NewLine}{ex.Message}");
        }
    }
}
```
___

#### Before applying the "fix".

The Release build encountered the issue, and successfully logged the detailed `ExecutionEngineException`.

```text
BEGIN PASS
[4/30/2024 11:58:48 AM] YVKHU
[4/30/2024 11:58:48 AM] NPGBK: ExecutionEngineException 
Attempting to JIT compile method 
'(wrapper delegate-invoke) void <Module>:invoke_callvirt_void_DBPI_string (IVSoftware.Portable.Quatch.Primitives.DBPI,string)' 
while running in aot-only mode. See https://docs.microsoft.com/xamarin/ios/internals/limitations for more information.
```

___

#### Applied the basic workaround in the **csproj** file.

```xml
<PropertyGroup Condition="'$(TargetFramework)' == 'net7.0-ios'">
	<UseInterpreter>True</UseInterpreter>
</PropertyGroup>
```

___

#### After applying the "fix".

The log now lists the query result, and the "SNRZ3" tag is further evidence a database query that successfully ran to completion.

```text
BEGIN PASS
[4/30/2024 12:50:17 PM] YVKHU
[4/30/2024 12:50:17 PM] 
Home
School
Storage Unit
Work
[4/30/2024 12:50:17 PM] SNRZ3
```

Hope this might be helpful.
___

#### Environment

Visual Studio 17.9.7
XCode 15.3
sqlite-net-pcl:  1.9.172
.NET 7

![image](https://github.com/praeclarum/sqlite-net/assets/17608266/c7d41f0f-ac1a-43c1-b268-cad17d2af894)

**Comment by SergTomcat** [comment]
Problem persists for many years. Is this project dead? Yet Microsoft put it in [documentation](https://learn.microsoft.com/en-us/dotnet/maui/data-cloud/database-sqlite?view=net-maui-9.0) still.

**Comment by phillippschmedt** [comment]
Can we bump this? It's a big issue for anyone using this project for maui with iOS! Thank you

**Comment by it11111111** [comment]
I dont think this project is maintained, yet Microsoft keeps listing it in their documentation.

**Comment by mikernet** [comment]
What are the best alternatives?
